src/Core/Query/LocalParameters/LocalParametersTrait.php

```diff
@@ -0,0 +1,222 @@
+ *
+ * @author wicliff <wicliff.wolda@gmail.com>
+ */
+trait LocalParametersTrait
+{
+    /**
+     * @var \Solarium\Core\Query\LocalParameters\LocalParameters
+     */
+    private $localParameters;
+
+    /**
+     * @return \Solarium\Core\Query\LocalParameters\LocalParameters
+     */
+    public function getLocalParameters(): LocalParameters
+    {
+        if (null === $this->localParameters) {
+            $this->localParameters = new LocalParameters();
+        }
+
+        return $this->localParameters;
+    }
+
+    /**
+     * Convenience method for backwards compatibility.
+     *
+     * @param string $exclude
+     *
+     * @throws \Solarium\Exception\OutOfBoundsException
+     *
+     * @return $this
+     */
+    public function addExclude(string $exclude): self
+    {
+        $this->getLocalParameters()->setExclude($exclude);
+
+        return $this;
+    }
+
+    /**
+     * Convenience method for backwards compatibility.
+     *
+     * @param array $excludes
+     *
+     * @throws \Solarium\Exception\OutOfBoundsException
+     *
+     * @return $this
+     */
+    public function setExcludes(array $excludes): self
+    {
+        $this->getLocalParameters()->clearExcludes()->addExcludes($excludes);
+
+        return $this;
+    }
+
+    /**
+     * Convenience method for backwards compatibility.
+     *
+     * @param array $excludes
+     *
+     * @throws \Solarium\Exception\OutOfBoundsException
+     *
+     * @return $this
+     */
+    public function addExcludes(array $excludes): self
+    {
+        $this->getLocalParameters()->addExcludes($excludes);
+
+        return $this;
+    }
+
+    /**
+     * Convenience method for backwards compatibility.
+     *
+     * @param string $exclude
+     *
+     * @throws \Solarium\Exception\OutOfBoundsException
+     *
+     * @return $this
+     */
+    public function removeExclude(string $exclude): self
+    {
+        $this->getLocalParameters()->removeExclude($exclude);
+
+        return $this;
+    }
+
+    /**
+     * Convenience method for backwards compatibility.
+     *
+     * @throws \Solarium\Exception\OutOfBoundsException
+     *
+     * @return $this
+     */
+    public function clearExcludes(): self
+    {
+        $this->getLocalParameters()->clearExcludes();
+
+        return $this;
+    }
+
+    /**
+     * Convenience method for backwards compatibility.
+     *
+     * @throws \Solarium\Exception\OutOfBoundsException
+     *
+     * @return array
+     */
+    public function getExcludes(): array
+    {
+        return $this->getLocalParameters()->getExcludes();
+    }
+
+    /**
+     * Init local parameters.
+     */
+    protected function initLocalParameters(): void
+    {
+        $this->localParameters = new LocalParameters();
+
+        foreach ($this->options as $name => $value) {
+            switch ($name) {
+                case 'exclude':
+                case LocalParameter::PARAMETER_MAP[LocalParameter::TYPE_EXCLUDE]:
```

Comment:
this list is rather long. What about collapsing the 'cases'?
```php
$this->getLocalParameters()->addValues($name, $value);
```

Reply:
i see your point, but that would mean the ``setValue($name, $value)``, ``addValues($name, $value)``, etc. methods have to be made public.

in my opinion this would "encourage" people to add values to the local parameters directly without solarium being able to enforce the actual method called or to (for now) have legacy option names set the correct local parameter or to have a different scenario for any given value (e.g. do the ``is_array()`` check and explode if necessary).

nevertheless, in order to simplify the switch i'd like to suggest to seperate the legacy parameter names ('exclude', 'tag', 'stats') and have them call the appropriate method until these options are deprecated, group the options which need an array check and group those remaining. make the addValues and addValues methods public and have them throw an exception when a parameter name is passed which is not in the ``LocalParameter::PARAMETER_MAP``. would that work for you?